modules/stochastic_tools/include/utils/LibtorchSimpleNeuralNet.h

```diff
@@ -0,0 +1,94 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#ifdef TORCH_ENABLED
+
+#include "LibtorchNeuralNetBase.h"
+
+#include "DataIO.h"
+
+namespace StochasticTools
+{
+
+// A class that describes a simple feed-forward neural net.
+class LibtorchSimpleNeuralNet : public LibtorchNeuralNetBase
+{
+public:
+  // Constructing using input parameters
+  LibtorchSimpleNeuralNet(std::string name,
+                          unsigned int num_inputs,
+                          std::vector<unsigned int> num_neurons_per_layer,
+                          unsigned int num_outputs);
+  // Virtual destructor
+  virtual ~LibtorchSimpleNeuralNet() {}
+
+  // Overriding the function from NeuralNetBase
+  virtual void addLayer(const std::string layer_name,
+                        const std::unordered_map<std::string, unsigned int> parameters) override;
+
+  // Overriding the forward sustitution function
+  torch::Tensor forward(torch::Tensor x);
+
+  const std::string & name() { return _name; }
+
+  const unsigned int & numInputs() { return _num_inputs; }
+
+  const unsigned int & numHiddenLayers() { return _num_hidden_layers; }
+
+  const std::vector<unsigned int> & numNeuronsPerLayer() { return _num_neurons_per_layer; }
+
+  const unsigned int & numOutputs() { return _num_outputs; }
```

Comment:
These methods should all be const and should have docstrings. And the unsigned ints should not be returned by reference (returning by ref for a small value implies that the underlying value changes, which it does not)

Reply:
Addressed!